src/utils/dev.ts

```diff
@@ -244,14 +244,15 @@ class NuxtDevServer extends EventEmitter {
       this.listener.server.on(
         'upgrade',
         async (req: any, socket: any, head: any) => {
-          if (
-            req.url.startsWith(
-              this._currentNuxt?.options.app.buildAssetsDir /* /_nuxt/ */,
-            )
-          ) {
+          const nuxt = this._currentNuxt
+          if (!nuxt) return
+          const viteHmrPath = (
+            nuxt.options.app.baseURL + nuxt.options.app.buildAssetsDir
+          ).replace(/\/\//g, '/')
+          if (req.url.startsWith(viteHmrPath)) {
             return // Skip for Vite HMR
           }
-          await this._currentNuxt?.server.upgrade(req, socket, head)
+          await nuxt.server.upgrade(req, socket, head)
```

Comment:
Was this change also needed as part of fix or simply refactor?

Reply:
It's a refactor to get rid of the `?` and also serves as a defensive guard to avoid we checking `undefined/undefined`